app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -1017,13 +1003,53 @@ class YearToDateLiabilityController @Inject() (
     }
   }
 
+  private def updateMandatoryEvidence(
+    currentAnswers: YearToDateLiabilityAnswers,
+    currentDraftReturn: DraftReturn,
+    currentJourney: FillingOutReturn,
+    upscanReference: String,
+    filename: String,
+    fileDescriptor: UpscanFileDescriptor
+  )(implicit request: RequestWithSessionData[_]): EitherT[Future, Error, Unit] = {
+    val newMandatoryEvidence = MandatoryEvidence(upscanReference, filename, fileDescriptor.timestamp)
+    val newAnswers = currentAnswers match {
+      case c: CalculatedYTDAnswers =>
+        c.fold(
+          _.copy(mandatoryEvidence = Some(newMandatoryEvidence), expiredEvidence = None),
+          _.copy(mandatoryEvidence = Some(newMandatoryEvidence))
+        )
+      case n: NonCalculatedYTDAnswers =>
+        n.fold(
+          _.copy(mandatoryEvidence = Some(newMandatoryEvidence), expiredEvidence = None),
+          _.copy(mandatoryEvidence = newMandatoryEvidence)
+        )
+    }
+    val newDraftReturn = currentDraftReturn.fold(
+      _.copy(yearToDateLiabilityAnswers = Some(newAnswers)),
+      _.copy(yearToDateLiabilityAnswers = Some(newAnswers))
+    )
+
+    for {
+      _ <- returnsService.storeDraftReturn(
+            newDraftReturn,
+            currentJourney.subscribedDetails.cgtReference,
+            currentJourney.agentReferenceNumber
+          )
+      _ <- EitherT(
+            updateSession(sessionStore, request)(
+              _.copy(journeyStatus = Some(currentJourney.copy(draftReturn = newDraftReturn)))
+            )
+          )
+    } yield ()
+  }
+
   @SuppressWarnings(Array("org.wartremover.warts.Any"))
```

Comment:
what's causing this - can't see anything in the code

Reply:
it's a knock-on effect from calling `handleGetFileDescriptorResult` in the for-comprehension